### ISSUE TITLE
Touch end doesn't fire outside of game window if hosted inside iframe

### ISSUE THREAD

**Issue by cesarvarela** [Scirra/Construct-3-bugs#3555]
## Problem description

When a game is hosted inside an iframe, and you press inside the game, move the cursor outside of the game window and release, the Touch End event never fires. IF the game is hosted inside an Iframe.

## Attach a .c3p

[touch end outside.c3p.zip](https://github.com/Scirra/Construct-3-bugs/files/3978870/touch.end.outside.c3p.zip)

## Steps to reproduce

1. Host the attached project inside an iframe 
2. Click inside the game window (browser console will log `start`)
3. Drag outside the game window and the iframe and release

## Observed result
The Touch End event won't fire

## Expected result
The Touch End event should fire

## More details

Code for the iframe:

```
<!DOCTYPE html>
<html>
<body>
	<iframe src="index.html" style="width: 600px; height: 900px;"></iframe>
</body>
</html>
```
## System details
- construct 3
- macOS Catalina

## Misc

I just tested it with the Mouse object and the behavior is similar, Button Released doens't fire when outside the iframe, without the iframe everything works properly.

But Instant Games run inside an iframe sadly

**Comment by shortercode** [comment]
If memory serves user input events are only allowed to go to a single frame, to prevent malicious frames from capturing information they shouldn't have.

**Comment by AshleyScirra** [comment]
Yeah, this is generally because the touchend (or mouseup) event goes to the parent page so the iframe never sees it. I'll take a look but there may not be any workaround. Does this only happen on Safari? It's useful to indicate if Chrome and Firefox are affected too.

**Comment by cesarvarela** [comment]
I've seen it in both safari and chrome

**Comment by AshleyScirra** [comment]
So this got more involved than I thought. It turns out Firefox and EdgeHTML work correctly. However there appears to be a browser bug with both Chrome and Safari where if we call preventDefault() on the starting pointer event, the pointer gets lost if it leaves the iframe.

If we don't call preventDefault() it works. However we have to call it in Safari to fix #3191, and we have to call it in Chrome to fix #1212. So it looks like we cannot work around it for the time being.

This looks like a browser bug - I don't see a reason the pointer should be completely lost because of calling preventDefault(). So I filed [an issue with Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=1035832)) and [an issue with Safari](https://bugs.webkit.org/show_bug.cgi?id=205446). Closing this issue as we will follow-up there.
